### ISSUE TITLE
Editor icons are not loading

### ISSUE THREAD

**Issue by geekdeb** [kolkov/angular-editor#475]
I am trying on **Angular-14** and **angular-editor-2.2.0-beta.0**. The Editor icons are not loading.
![Screenshot from 2022-07-12 01-53-24](https://user-images.githubusercontent.com/81557373/178351835-8fab8c0b-9d64-49e7-b525-b75f1e98c4e9.png)

Also, I could see these errors in the console. And I guess this is due to the removal of the CDN link in the latest merge. 
![image](https://user-images.githubusercontent.com/81557373/178351761-149c93bc-b720-4c0c-a76c-89e7cdb9eb5c.png)

Kindly let me know the workaround if any.

**Comment by pietzschke** [comment]
Yes, please use the new v3.0.0-beta.0 version and include the following in the assets section in the angular.json:
```
{
  "glob": "**/*",
  "input": "./node_modules/@kolkov/angular-editor/assets/",
  "output": "./assets/fonts/"
}
```

**Comment by alex-goods** [comment]
hi
i tried this approach but its still not working, i have no console errors but the icons are not showing, i can see them in the devtools but the button is empty
![image](https://user-images.githubusercontent.com/113447819/189918213-235318c6-2f88-47b7-afbf-d7d88cb21fce.png)
![image](https://user-images.githubusercontent.com/113447819/189918380-6e5c9aed-634c-49fd-9ed6-d18366502fe6.png)

**Comment by anandjaisy** [comment]
```
"assets": [
              "src/favicon.ico",
              "src/assets",
              {
                "glob": "**/*",
                "input": "./node_modules/@kolkov/angular-editor/assets/",
                "output": "./assets/fonts/"
              }
            ],
```

worked for me

**Comment by Tatukis** [comment]
Have the same issue.

![image](https://user-images.githubusercontent.com/12802537/200616554-17717a0a-11de-4edc-b666-5c9fcfcca136.png)

**Comment by ArturGudiev** [comment]
There are 2 assets. build and test in angular.json 

Make sure you included it into build assets

**Comment by Tatukis** [comment]
> There are 2 assets. build and test in angular.json
> 
> Make sure you included it into build assets

I fix the issue installing awesome fonts or something like that. I couldn't find the dependency in the documentation.

**Comment by alex-goods** [comment]
> anandjaisy

this worked for me too, thx:>

**Comment by philecom** [comment]
If the above solutions don't work for you, you can try what I did.
1) Copy the entire content of "node_modules\@kolkov\angular-editor\assets"
2) Create a folder called "fonts" in your app assets folder so you will have "src\assets\fonts"
3) Paste the copied contents (font awesome files).

that's all.

**Comment by youquijano** [comment]
> ```
> "assets": [
>               "src/favicon.ico",
>               "src/assets",
>               {
>                 "glob": "**/*",
>                 "input": "./node_modules/@kolkov/angular-editor/assets/",
>                 "output": "./assets/fonts/"
>               }
>             ],
> ```
> 
> worked for me

this also worked for me.

**Comment by dmitrikovalevski** [comment]
I moved object with { glob, input, output } above. It take first position below all strings. This works for me.


![image](https://user-images.githubusercontent.com/81647895/235158360-7ca96713-b96f-4597-857f-35f668b9ba62.png)

**Comment by veer05** [comment]
Same issue, I use --base-href and the application is trying to look at parent path and fails
Temporary workaround

In index.html file add this
`<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css"> 
`2. In your css file add this

```
::ng-deep angular-editor-toolbar button i {
  font-family: FontAwesome !important; 
  } 
```

**Comment by Pimpollo30** [comment]
> Same issue, I use --base-href and the application is trying to look at parent path and fails Temporary workaround
> 
> In index.html file add this `<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css"> `2. In your css file add this
> 
> ```
> ::ng-deep angular-editor-toolbar button i {
>   font-family: FontAwesome !important; 
>   } 
> ```

@veer05 temporary solution worked for me, thanks

**Comment by ZenulAbidin** [comment]
This might be obvious but also check that your font was not set to a white color by accident by some framework - in that case the fonts are actually loaded but you might be fooled into thinking they are not present.

**Comment by BraianS** [comment]
Good morning, @T4toh @ArturGudiev . Thank you. Worked perfectly.

**Comment by qqaimh** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by BraianS** [comment]
Good morning everyone, it worked perfectly.

**Comment by mohaseebtariq** [comment]
https://github.com/kolkov/angular-editor/pull/573

I fixed this by adding the font, and it worked initially. But when I added a glob in angular.json to include icons in my build, they still wouldn’t show up when running locally. Another issue was that my app has a baseUrl, so all assets load from there, but the FontAwesome files were being served internally from the library, which broke the pathing. To solve this for good, I forked and added the Angular @fortawesome/angular-fontawesome package. This way, the icons load consistently with the rest of the assets, and the issue won’t reoccur.

**Comment by Farzane2630** [comment]
I had the same issue and this worked for me:

add this object in **angular.json** file in assets: (as it is in the docs)
`
{
  "glob": "**/*",
  "input": "node_modules/@kolkov/angular-editor/assets/icons",
  "output": "assets/ae-icons/"
}
`

and add this style in **styles.scss** file:

`
::ng-deep angular-editor-toolbar button i {
  font-family: FontAwesome !important;
  color: black !important;
}
`

**Comment by mohaseebtariq** [comment]
::ng-deep got deprecated so if you are using later versions that fixed may not work @Farzane2630

**Comment by Farzane2630** [comment]
@mohaseebtariq 
Thank you for your comment.
 I edited my answer.
Though the previous worked well with Angular18

**Comment by durgeshsamcomtechnobrains** [comment]
@Farzane2630 can you please tell me how you implement in Angular 18

![Image](https://github.com/user-attachments/assets/4bd23824-8fee-4d26-a6ea-95c60d984a3c)

![Image](https://github.com/user-attachments/assets/8e189434-a4ec-4adf-b5e0-64e601882a9e)

![Image](https://github.com/user-attachments/assets/7d6a98c0-213a-4ab4-844d-584e8505b722)


Please help me to solve this issue in Angular 18
- Angular 17 
- it's working fine
- 
![Image](https://github.com/user-attachments/assets/8e22e049-5b91-43c4-93bd-4d8c34c55231)

![Image](https://github.com/user-attachments/assets/0aec18d0-9806-4ba5-a934-64d96115278c)

**Comment by mohaseebtariq** [comment]
@durgeshsamcomtechnobrains when you build the project do you see the font files in the build folder?

**Comment by AravindM-Contus** [comment]
@durgeshsamcomtechnobrains , Could you provide me the solution to fix the issue?

**Comment by durgeshsamcomtechnobrains** [comment]
@AravindM-Contus you just need to change in 2 files 
1.src/styles.scss
2. angular.json

![Image](https://github.com/user-attachments/assets/62002896-1478-4987-b9d4-ddc255eb2630)

**Comment by meetitexpert** [comment]
> [@AravindM-Contus](https://github.com/AravindM-Contus) you just need to change in 2 files 1.src/styles.scss 2. angular.json
> 
> ![Image](https://github.com/user-attachments/assets/62002896-1478-4987-b9d4-ddc255eb2630)

Hi, I am facing an issue. This change works on localhost, but when I deploy the build, it does not work in production. Has anyone faced this issue? 

I need help.

<img width="632" alt="Image" src="https://github.com/user-attachments/assets/c94e8283-ff4c-4860-8d37-6c7145108796" />

**Comment by prashan1** [comment]
@meetitexpert 
https://www.npmjs.com/package/@kolkov/angular-editor#:~:text=Styling,themes/default.scss%22%0A%5D%2C
follow this, add these lines in your main css file, should work then.

**Comment by kolkov** [comment]
Closed: Icon loading issues resolved in v3.0+ with embedded SVG icons. No external assets needed. Upgrade to @kolkov/angular-editor@3.0.2
